src/openassetio-core/src/hostApi/Manager.cpp

```diff
@@ -53,12 +55,11 @@ std::optional<Str> entityReferencePrefixFromInfo(const log::LoggerInterfacePtr &
   if (auto iter = info.find(Str{constants::kInfoKey_EntityReferencesMatchPrefix});
       iter != info.end()) {
     if (const auto *prefixPtr = std::get_if<openassetio::Str>(&iter->second)) {
-      std::string msg = "Entity reference prefix '";
-      msg += *prefixPtr;
-      msg +=
-          "' provided by manager's info() dict. Subsequent calls to isEntityReferenceString will"
-          " use this prefix rather than call the manager's implementation.";
-      logger->debugApi(msg);
+      logger->debugApi(
+          fmt::format("Entity reference prefix '{}' provided by manager's info() dict. Subsequent"
+                      " calls to isEntityReferenceString will use this prefix rather than call the"
+                      " manager's implementation.",
+                      *prefixPtr));
```

Comment:
That's an interesting idea, so logger methods become a wrapper around `fmt::format`, and we can skip the interpolation if the log severity is too low. :thinking: tricky actually, since we can't use variadic templates for a `virtual` method.  Worth looking into in case there's a way to do it.

Reply:
A macro, or even a normal function, that wraps the `LoggerInterface` as a middleware, could do it I think, yeah.

Reply:
Ace!